pkg/pool-linear/contracts/LinearPoolRebalancer.sol

```diff
@@ -77,13 +93,55 @@ abstract contract LinearPoolRebalancer {
         (uint256 mainTokenBalance, , , ) = _vault.getPoolTokenInfo(_poolId, _mainToken);
 
         if (mainTokenBalance < desiredMainTokenBalance) {
-            _rebalanceLackOfMainToken(desiredMainTokenBalance - mainTokenBalance, recipient);
-        } else {
-            _rebalanceExcessOfMainToken(mainTokenBalance - desiredMainTokenBalance, recipient);
+            if (extraMain > 0) {
+                _receiveExtraMainDuringLackOfMain(mainTokenBalance, extraMain);
+            }
+
+            return _rebalanceLackOfMainToken(desiredMainTokenBalance - mainTokenBalance, recipient);
+        } else if (mainTokenBalance > desiredMainTokenBalance) {
+            if (extraMain > 0) {
+                _receiveExtraMainDuringExcessOfMain(mainTokenBalance, extraMain);
+            }
+
+            return _rebalanceExcessOfMainToken(mainTokenBalance - desiredMainTokenBalance, recipient);
+        }
+    }
+
+    function _receiveExtraMainDuringLackOfMain(uint256 mainTokenBalance, uint256 extraMain) private {
+        // The Pool lacks main tokens, but if the main balance is above the lower target (i.e. it is still in the no-fee
+        // zone), then there's no accumulated fees. It is therefore possible that the amount of wrapped tokens the Pool
+        // will give us in exchange for main cannot be unwrapped into the amount of main tokens desired by the Pool, due
+        // to  the Pool choosing to round in its favor (resulting in an unrealistic wrap/unwrap rate).
+        //
+        // This is typically a non-issue as the Linear Pool accumulated fees more than cover for said rounding error,
+        // but in scenarios with no fees we require some extra main tokens to be able to make the Pool whole. Any excess
+        // will be sent at the end of the transaction to the recipient.
+
+        (uint256 lowerTarget, ) = _pool.getTargets();
+        // The lower target is upscaled, so we undo that
+        if (mainTokenBalance > FixedPoint.divDown(lowerTarget, _mainTokenScalingFactor)) {
```

Comment:
```suggestion
        if (mainTokenBalance >= FixedPoint.divDown(lowerTarget, _mainTokenScalingFactor)) {
```

If we're right on the edge of the no-fee zone there's still no fees to cover the rounding errors so we should be inclusive here.